cvmfs/swissknife_graft.cc

```diff
@@ -0,0 +1,249 @@
+/**
+ * This file is part of the CernVM File System.
+ *
+ * Process a set of input files and create appropriate graft files.
+ */
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+#include <cstdio>
+#include <iostream>
+#include <vector>
+#include <sstream>
+
+#include "compression.h"
+#include "fs_traversal.h"
+#include "hash.h"
+
+#include "swissknife_graft.h"
+
+#define CHUNK_SIZE 16*1024
+
+int swissknife::CommandGraft::ChecksumFd(int fd, shash::Any &id, off_t &processed_size) {  // NOLINT
```

Comment:
Oh - and `CompressFd2Null()` has a nasty bug: `result` is initialized to -1 (for the error case) but the function return is a `bool`.

Hence, it'll return `true` on either success or failure.  I'll fix that too.